src/backport.ts

```diff
@@ -86,19 +90,48 @@ export class Backport {
 
       let commitShasToCherryPick: string[];
 
-      // find out if "squashed and merged" or "rebased and merged"
-      if (await this.github.isSquashed(mainpr) || mainpr.commits == 1) {
-        console.log("PR was squashed and merged");
-        // if squashed, then use the merge_commit_sha
-        commitShasToCherryPick = [
-          await this.github.getMergeCommitSha(mainpr),
-        ]?.filter(Boolean) as string[];
-      } else {
-        // if rebased, then use all the commits from the original PR
-        console.log("PR was rebased and merged");
-        commitShasToCherryPick = commitShas;
+      /*
+
+      squashed:
+        - > single parent
+        - > the parent of the `merge_commit_sha` is NOT associated with the PR
+
+        => use merge_commit_sha
+
+      rebased single-commit:
+        - > single parent
+        - > the parent of the `merge_commit_sha` is associated with the PR
+
+        => use commits associated with the PR
+
+      rebased multi-commit:
+        - > single parent,
+        - > the parent of the `merge_commit_sha` is associated with the PR
+
+        => use commits associated with the PR
+
+      merge-commit:
+        - > multiple parents
+
+        => use commits associated with the PR
```

Comment:
For a rebased single-commit, the parent of the `merge_commit_sha` is sadly not associated with the PR.

💡 We can differentiate between a squash and a rebase by the number of commits on the pull request:
- if there are more than one commit on the pull request, and the parent of the `merge_commit_sha` is associated with the pull request -> it was rebased
- if there are more than one commit on the pull request, and the parent of the `merge_commit_sha` is not associated with the pull request -> it was squashed
- if there is one commit on the pull request -> it was rebased (only multiple commits can be squashed)

Combine it with the merge commit and we get quite simple logic:
- if the `merge_commit_sha` has multiple parents -> it was merged as a merge commit
- if there is more than one commit on the pull request, and the parent of the `merge_commit_sha` is associated with the pull request -> it was rebased
- if there is more than one commit on the pull request, and the parent of the `merge_commit_sha` is not associated with the pull request -> it was squashed
- if there is one commit on the pull request -> it was rebased

Reply:
It looks like we're getting closer though. We've agreed on:

* if the merge_commit_sha has multiple parents -> it was merged as a merge commit
* if there is more than one commit on the pull request, and the parent of the merge_commit_sha is associated with the pull request -> it was rebased
* if there is more than one commit on the pull request, and the parent of the merge_commit_sha is not associated with the pull request -> it was squashed

> if there is one commit on the pull request -> it was rebased (only multiple commits can be squashed)

You can also squash just one commit :( People use it to rewrite the commit message. 

By looking at just this one condition we can't reliably tell if it was squashed or rebased _but_ we can choose to use the `merge_commit_sha` as it works for both cases. So to follow our ruleset.


* if there is one commit on the pull request -> it was either rebased or squashed. We can use the `merge_commit_sha`

Reply:
>People use it to rewrite the commit message.

💭 That's an interesting point

>if there is one commit on the pull request -> it was either rebased or squashed. We can use the `merge_commit_sha`

👍 I like this approach! Let's go for it!

Reply:
check -> https://github.com/korthout/backport-action/pull/399/commits/edf37b4ab9f2ff647d0198a1ebf271eb38ce56e0

Reply:
The switch case needs `break`. Fixing